src/backend/distributed/deparser/citus_ruleutils.c

```diff
@@ -390,6 +392,54 @@ pg_get_tableschemadef_string(Oid tableRelationId, IncludeSequenceDefaults
 			}
 #endif
 
+			if (attributeForm->attidentity && includeIdentityDefaults)
+			{
+				Oid seqOid = getIdentitySequence(RelationGetRelid(relation),
+												 attributeForm->attnum, true);
+				Assert(seqOid != InvalidOid);
+
+				char *sequenceName = generate_qualified_relation_name(seqOid);
+
+				if (includeIdentityDefaults == INCLUDE_IDENTITY_AS_SEQUENCE_DEFAULTS)
+				{
+					if (pg_get_sequencedef(seqOid)->seqtypid != INT8OID)
+					{
+						appendStringInfo(&buffer,
+										 " DEFAULT worker_nextval(%s::regclass)",
+										 quote_literal_cstr(sequenceName));
+					}
+					else
+					{
+						appendStringInfo(&buffer, " DEFAULT nextval(%s::regclass)",
+										 quote_literal_cstr(sequenceName));
+					}
+				}
+				else if (includeIdentityDefaults == INCLUDE_IDENTITY)
+				{
+					Form_pg_sequence pgSequenceForm = pg_get_sequencedef(seqOid);
+					uint64 sequenceStart = nextval_internal(seqOid, false);
+					char *sequenceDef = psprintf(
+						" GENERATED %s AS IDENTITY (INCREMENT BY " INT64_FORMAT \
+						" MINVALUE " INT64_FORMAT " MAXVALUE "
+						INT64_FORMAT \
+						" START WITH " INT64_FORMAT " CACHE "
+						INT64_FORMAT " %sCYCLE)",
+						attributeForm->attidentity == ATTRIBUTE_IDENTITY_ALWAYS ?
+						"ALWAYS" : "BY DEFAULT",
+						pgSequenceForm->seqincrement,
+						pgSequenceForm->seqmin,
+						pgSequenceForm->seqmax, sequenceStart,
+						pgSequenceForm->seqcache,
+						pgSequenceForm->seqcycle ? "" : "NO ");
+
+					appendStringInfo(&buffer, "%s", sequenceDef);
+				}
+				else
+				{
+					Assert(false);
```

Comment:
You already exhaust all the possible values of the `IncludeIdentityDefaults` type so this `Assert` is non-reachable, hence shouldn't be here.

Maybe you can use `else` instead of `else if` above and explain in the comments that the only value left is `INCLUDE_IDENTITY`.